h2o-genmodel/src/main/java/hex/genmodel/algos/gam/GamMojoModelBase.java

```diff
@@ -83,8 +81,8 @@ private void imputeMissingWithMeans(double[] data) {
       if (Double.isNaN(data[ind])) data[ind] = _catNAFills[ind];
 
     if (data.length == nfeatures()) { // using centered gam cols, nfeatures denotes centered gam columns
-      for (int ind = 0; ind < _numsCenter; ind++)
-        if (Double.isNaN(data[ind + _cats])) data[ind] = _numNAFillsCenter[ind];
+      for (int ind = _cats; ind < _numsCenter + _cats; ind++)
```

Comment:
This is the actual **fix** for the "NaN problem". The imputation of categoricals has been done in a correct way, but the imputation of the rest was started using bad index - it overrode existing categoricall imputation and did not reach the end. Therefore, the fix is only in indexing.

I took a look at how `DataInfo.extractDenseRow` does it (actually used for GAM native model scoring) - looks the same to me.

cc: @wendycwong @michalkurka @honzasterba